File: DDD DSL/DDD DSL Design War FizzBuzzWhizz Reloaded/Code/Java/src/fayelab/ddd/fbwreloaded/complete/compiler/oo/FizzBuzzWhizz.java
package fayelab.ddd.fbwreloaded.complete.compiler.oo;

import java.util.stream.Collectors;
import java.util.stream.IntStream;

import static java.util.Arrays.asList;
import static fayelab.ddd.fbwreloaded.complete.compiler.oo.SpecTool.spec;
import static fayelab.ddd.fbwreloaded.complete.compiler.oo.Compiler.compile;

public class FizzBuzzWhizz
{
    public void run()
    {
        output(runSpec(spec()));
    }
    
    public void run(String progFileName)
    {
        output(runSpec(Parser.parse(progFileName)));
    }
    
    private String runSpec(Rule spec)
    {
        return IntStream.rangeClosed(1, 100)
                        .mapToObj(n -> asList(String.valueOf(n), compile(spec).apply(n).get()))
                        .map(pair -> String.join(" -> ", pair.get(0), pair.get(1)))
                        .collect(Collectors.joining("\n"));
    }
    
    private void output(String result)
    {
        System.out.println(result);
    }

    public static void main(String[] args)
    {
        new FizzBuzzWhizz().run("prog_1.fbw");
    }
}

File: DDD DSL/DDD DSL Design War FizzBuzzWhizz Reloaded/Code/Python/complete/compiler/oo/fizzbuzzwhizz.py
import sys

from fbwparser.parser import *
from compiler import *

def run(prog_file_name):
    prog_file_full_name = ''.join([sys.path[0], '/scripts/', prog_file_name])
    _output(_run_spec(parse(prog_file_full_name)))

def _run_spec(s):
    return [(n, compile(s).apply(n)) for n in range(1, 101)]

def _output(results):
    for result in results:
        print ' -> '.join([str(result[0]), result[1][1]])

if __name__ == '__main__':
    run('prog_1.fbw')